### PR TITLE
- Turn on new calls and use shrU

### PR DESCRIPTION
**Pull request by apcraig**
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Turn on new code
- [X] Developer(s): 
    apcraig, jflemieux
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    bit-for-bit until usage of calc_shearT_DeltaT
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit except grid C
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [X] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [X] No 
- [X] Please provide any additional information or relevant details below:

- turn on new calls and use shrU
- bit-for-bit except for when using calc_shearT_DeltaT
- fix some bit-for-bit issues from earlier PRs in cheyenne+intel non-debug compiler optimization

### CODE REVIEW

**Comment by apcraig** [comment]
I am running a full test suite overnight on cheyenne.

There is a CPP in ice_dyn_evp.F90 to recover the prior results bit-for-bit.  See line 1652.  At present, this is set to run the new code.

Still lots of code cleanup to do, as well as computation reductions for the C grid.

File: cicecore/cicedynB/dynamics/ice_dyn_evp.F90 [hunk 1]
```diff
@@ -1798,6 +1805,19 @@ subroutine stressC_U   (nx_block,   ny_block,  &
...
+                                 uvelN,      vvelN,     &
+                                 uvelU,      vvelU,     &
+                                 dxE,        dyN,       &
+                                 dxU,        dyU,       &
+                                 ratiodxN,   ratiodxNr, &
+                                 ratiodyE,   ratiodyEr, &
+                                 epm,  npm,  uvm,       &
+                                 divU,       tensionU,  &
+                                 shearU,     DeltaU     )
+            shearU = shrU(i,j)
```
Comment (JFLemieux73) [inline cicecore/cicedynB/dynamics/ice_dyn_evp.F90:1820]: Will we remove all the shearU later and only have shrU?

Comment (apcraig) [reply cicecore/cicedynB/dynamics/ice_dyn_evp.F90:1820]: Yes.

**Review by dabail10 — APPROVED** [review]

**Comment by apcraig** [comment]
Test Results, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#6c859d90eb78fc5ef5ee98885486791a2114c176

**Pull request merged**